### PR TITLE
Add physics interpolation docs to 4.3

### LINKED ISSUE #95936: 2D physics interpolation ghosted sprite2d
### Tested versions

4.3 official

### System information

win10

### Issue description

Enable physical interpolation

Turn off physical interpolation

### Steps to reproduce

See example project

### Minimal reproduction project (MRP)

[t.zip](https://github.com/user-attachments/files/16707007/t.zip)

### PR DESCRIPTION
**Pull request by skyace65**
This PR adds the physics interpolation docs from 3.6 to the 4.3 branch. This is mostly a copy and paste with the following changes:

- The camera example from one page was removed as it was a 3D example.
- the bottom of the 2D and 3D page was mostly removed. it now just states that interpolation isn't supported for particles.
- The existing images were converted to webp, and the one picture of the editor was re-done, all images have gone through webp compression.
- Notes added on each page stating that physics interpolation does not work in 3D
- Removed info on MultiMesh from the advanced physics interpolation page since those methods don't exist in 4.3
- Updated the project settings links for physics FPS since the setting was renamed in 4.x
- Changed a reference for camera (3D camera) to "a node" (line 151 on using physics interpolation)

I'm keeping the 2D and 3D page mostly as is because the 2D information on that page is still useful, it would be time consuming to remove everything 3D related, and it would seem mildly pointless since the 3D information will be relevant again in a future update. Though I'm open to doing more changes if anyone thinks it's necessary.

### CODE REVIEW

**Comment by skyace65** [comment]
@rburing Could the 3D camera example in these docs be re-worked for a 2D scene? And if so how heavily would it have to be modified? I'm trying to figure out if it should be altered or just removed.

**Comment by rburing** [comment]
@skyace65 I would remove all of it for 4.3 for simplicity.

Also there is no `get_global_transform_interpolated` or interpolation of 2D particles in 4.3.

We can add back the camera example for 4.4 when 3D physics interpolation is merged.

cc @lawnjelly

**Comment by skyace65** [comment]
@rburing it's ready for review now.

File: tutorials/physics/interpolation/2d_and_3d_physics_interpolation.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
+.. _doc_2d_and_3d_physics_interpolation:
+
+2D and 3D physics interpolation
+===============================
+
+Generally 2D and 3D physics interpolation work in very similar ways. However,
+there are a few differences, which will be described here.
+
+.. note:: currently only 2D physics interpolation works in Godot.
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/2d_and_3d_physics_interpolation.rst:9]:
```suggestion
.. note:: Currently only 2D physics interpolation works in Godot.
```

File: tutorials/physics/interpolation/2d_and_3d_physics_interpolation.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- However this means that in 3D, pivots that occur in the ``SceneTree``
+  (due to parent child relationships) can only be interpolated
+  **approximately** over the physics tick. In most cases this will not
+  matter, but in some situations the interpolation can look slightly *off*.
+- In 2D, interpolated local transforms are passed down to children during
+  rendering. This means that if a parent is set to
+  ``physics_interpolation_mode`` ``On``, but the child is set to ``Off``,
+  the child will still be interpolated if the parent is moving. *Only the
+  child's local transform is uninterpolated.* Controlling the on / off
+  behaviour of 2D nodes therefore requires a little more thought and planning.
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/2d_and_3d_physics_interpolation.rst:38]:
```suggestion
  behavior of 2D nodes therefore requires a little more thought and planning.
```
For all

File: tutorials/physics/interpolation/advanced_physics_interpolation.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
+.. _doc_advanced_physics_interpolation:
+
+Advanced physics interpolation
+==============================
+
+Although the previous instructions will give satisfactory results in a lot of
+games, in some cases you will want to go a stage further to get the best
+possible results and the smoothest possible experience.
+
+.. note:: currently only 2D physics interpolation works in Godot.
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/advanced_physics_interpolation.rst:10]:
```suggestion
.. note:: Currently only 2D physics interpolation works in Godot.
```

File: tutorials/physics/interpolation/physics_interpolation_introduction.rst [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+The most obvious solution is to get rid of the problem, by ensuring there is
+a physics tick that coincides with every frame. This used to be the approach
+on old consoles and fixed hardware computers. If you know that every player
+will be using the same hardware, you can ensure it is fast enough to
+calculate ticks and frames at e.g. 50 FPS, and you will be sure it will work
+great for everybody.
+
+However, modern games are often no longer made for fixed hardware. You will
+often be planning to release on desktop computers, mobiles and more, all of
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/physics_interpolation_introduction.rst:57]:
```suggestion
often be planning to release on desktop computers, mobiles, and more. All of
```
I'd break this long sentence

File: tutorials/physics/interpolation/physics_interpolation_introduction.rst [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+.. image:: img/fti_graph_interpolated.webp
+
+Linear interpolation
+^^^^^^^^^^^^^^^^^^^^
+
+The simplest way to achieve this is linear interpolation, or lerping, which
+you may have used before.
+
+.. note:: currently only 2D physics interpolation works in Godot.
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/physics_interpolation_introduction.rst:117]:
```suggestion
.. note:: Currently only 2D physics interpolation works in Godot.
```

File: tutorials/physics/interpolation/physics_interpolation_quick_start_guide.rst [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+          3D interpolation is expected to come in a future update.
+
+- Turn on physics interpolation: :ref:`ProjectSettings.physics/common/physics_interpolation<class_ProjectSettings_property_physics/common/physics_interpolation>`
+- Make sure you move objects and run your game logic in
+  ``_physics_process()`` rather than ``_process()``. This includes moving
+  objects directly *and indirectly* (by e.g. moving a parent, or using
+  another mechanism to automatically move nodes).
+- Be sure to call :ref:`Node.reset_physics_interpolation<class_Node_method_reset_physics_interpolation>`
+  on nodes *after* you first position or teleport them, to prevent
+  "streaking"
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/physics_interpolation_quick_start_guide.rst:16]:
```suggestion
  "streaking".
```

File: tutorials/physics/interpolation/using_physics_interpolation.rst [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+Using physics interpolation
+===========================
+How do we incorporate physics interpolation into a Godot game? Are there any
+caveats?
+
+We have tried to make the system as easy to use as possible, and many
+existing games will work with few changes. That said there are some
+situations which require special treatment, and these will be described.
+
+.. note:: currently only 2D physics interpolation works in Godot.
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/using_physics_interpolation.rst:12]:
```suggestion
.. note:: Currently only 2D physics interpolation works in Godot.
```

File: tutorials/physics/interpolation/using_physics_interpolation.rst [hunk 8]
```diff
@@ -0,0 +1,159 @@
+.. _doc_using_physics_interpolation:
+
+Using physics interpolation
+===========================
+How do we incorporate physics interpolation into a Godot game? Are there any
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/using_physics_interpolation.rst:5]:
```suggestion

How do we incorporate physics interpolation into a Godot game? Are there any
```

File: tutorials/physics/interpolation/using_physics_interpolation.rst [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+longer limited to values that are multiples of the user's monitor refresh
+rate (for stutter-free gameplay if the target FPS is reached).
+
+As a rough guide:
+
+.. csv-table::
+    :header: "Low tick rates (10-30)", "Medium tick rates (30-60)", "High tick rates (60+)"
+    :widths: 20, 20, 20
+    
+    "Better CPU performance","Good physics behaviour in complex scenes","Good with fast physics"
```
Comment (AThousandShips) [inline tutorials/physics/interpolation/using_physics_interpolation.rst:89]:
```suggestion
    "Better CPU performance","Good physics behavior in complex scenes","Good with fast physics"
```
For all here too

**Review by AThousandShips — COMMENTED** [review]
Will go over again for more details but these were the immediate

**Comment by rburing** [comment]
I would really remove all details about 3D, since nothing is merged on that front. Minimal docs are fine to start with. I'll be glad to update the docs for 3D when the time comes, but at the moment I'm on vacation.

File: tutorials/physics/interpolation/physics_interpolation_introduction.rst [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+This can make quality assurance difficult with hard to reproduce bugs,
+especially in AAA games where problems of this sort can be very costly. This
+can also be problematic for multiplayer games for competitive integrity, as
+running the game at certain tick rates may be more advantageous than others.
+
+Lock the tick rate, but use interpolation to smooth frames in between physics ticks
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This has become one of the most popular approaches to dealing with the
+problem. It is supported by Godot 3.5 and later in 3D (although it is
```
Comment (lawnjelly) [inline tutorials/physics/interpolation/physics_interpolation_introduction.rst:88]: Make relevant to 4.x.

**Comment by lawnjelly** [comment]
Agree with @rburing on simply removing the details of 3D, especially in the `2D and 3D physics interpolation` section. I actually only added that section *after* adding 2D to emphasize the differences.

So perhaps just titling it `2D physics interpolation` for now and only including 2D info would be good.

**Comment by tetrapod00** [comment]
So, 3D physics interpolation (https://github.com/godotengine/godot/pull/92391) was merged into 4.4.

Is it still the right approach to start with this PR targeting 4.3 that only discusses 2D and removes all mentions of 3D? There are notes about the lack of 3D support that will only be relevant for a single minor version.

**Pull request closed without merging**